src/redux/reducers/index.js

```diff
@@ -0,0 +1,40 @@
+import {
+    FETCH_DEFAULT,
+    FETCH_FROM_SEARCH,
+    CHANGE_SEARCH,
+    CHANGE_SORTING, UPDATE_SEARCH_VALUE, SELECT_MOVIE, UN_SELECT_MOVIE
+} from '../constants/action-types';
+import {sortingTypeForSearch, TITLE} from "../../components/constants/CommonConstants";
+
+const initialState = {
+    movies: [],
+    sortingType: sortingTypeForSearch["release date"],
+    searchOption: TITLE,
+    selectedMovie: '',
+    isSearching: false,
+    error: false,
+    searchValue: ''
+};
+
+function rootReducer(state = initialState, action) {
+    switch (action.type) {
+        case FETCH_DEFAULT:
+            return { ...state, movies: action.payload };
+        case FETCH_FROM_SEARCH:
+            return { ...state, movies: action.payload };
+        case CHANGE_SEARCH:
+            return { ...state, searchOption: action.payload };
+        case CHANGE_SORTING:
+            return { ...state, sortingType: action.payload };
+        case UPDATE_SEARCH_VALUE:
+            return { ...state, searchValue: action.payload };
+        case SELECT_MOVIE:
+            return { ...state, selectedMovie: action.payload };
+        case UN_SELECT_MOVIE:
+            return { ...state, selectedMovie: '' };
+        default:
+            return state;
```

Comment:
Lets separate here movies and search reducers. Please don't care about selected movie, as it will go away after router applying.
In our example movies reducer will be used for searched movies and similar movies. So we can make Results smarter and connect it to store.

Reply:
divided into 2 child reducers